draft-ietf-quic-transport.md

```diff
@@ -4125,37 +4172,38 @@ instead, receive streams track the delivery of data to the application or
 application protocol some of which cannot be observed by the sender.
 
 ~~~
-       o
-       | Recv STREAM / STREAM_BLOCKED / RST_STREAM
-       | Create Bidirectional Stream (Sending)
-       | Recv MAX_STREAM_DATA
-       | Create Higher-Numbered Stream
-       v
-   +-------+
-   | Recv  | Recv RST_STREAM
-   |       |-----------------------.
-   +-------+                       |
-       |                           |
-       | Recv STREAM + FIN         |
-       v                           |
-   +-------+                       |
-   | Size  | Recv RST_STREAM       |
-   | Known +---------------------->|
-   +-------+                       |
-       |                           |
-       | Recv All Data             |
-       v                           v
-   +-------+                   +-------+
-   | Data  | Recv RST_STREAM   | Reset |
-   | Recvd +<-- (optional) --->| Recvd |
-   +-------+                   +-------+
-       |                           |
-       | App Read All Data         | App Read RST
-       v                           v
-   +-------+                   +-------+
-   | Data  |                   | Reset |
-   | Read  |                   | Read  |
-   +-------+                   +-------+
+                       o
+                       | Recv STREAM / STREAM_BLOCKED / RST_STREAM
+                       | Create Bidirectional Stream (Sending)
+                       | Recv MAX_STREAM_DATA
+                       | Create Higher-Numbered Stream
+                       v
+                   +-------+ Send STOP_SENDING
+                 .-|       |-------------------------------.
+Send             | | Recv  |                               |
+ MAX_STREAM_SIZE '>|       | Recv RST_STREAM               |
+                   +-------+-----------------------.       |
+                       |                           |       |
+                       | Recv STREAM + FIN         |       |
+                       v                           |       v
+                   +-------+                       |   +-------+
+                   | Size  |                       |   | Wait  |
+                   | Known + Recv RST_STREAM       |   | STOP  |
+                   +-------+---------------------->|   +-------+
```

Comment:
What is this `Wait STOP` state?

Reply:
Presumably the state where incoming data is being discarded, but the solicited RST_STREAM hasn't been received yet.  We originally said that STOP_SENDING didn't change the state, just hinted to the other side that it should do so.

I don't object to making this a defined state, but if so, then the description needs to be added below as well.  Let's make this a separate PR.

Reply:
i initially though about transition to "Data Recvd" on reception of STOP_SENDING but that would not capture that you are actually waiting for a RST_STREAM. 

The text is mostly there it only needs to name the new state if people agree.

Reply:
That sort of change has to be a separate PR.